packages/react-native-windows-init/src/Cli.ts

```diff
@@ -210,79 +208,39 @@ function getMatchingReactNativeSemVerForReactNativeWindowsVersion(
-      // Ideally we'd be able to just use npm.packages.range(pkg, versionSemVer) here,
-      // but alas it fails to give us the right result for react-native-windows@^0.60.0-0
-      // as it fails to return pre-release versions
-      npm.packages.releases(
-        pkg,
-        (err: any, details: {[key: string]: Record<string, any>} | null) => {
-          if (err) {
-            reject(err);
-          } else if (details) {
-            const versions = Object.keys(details);
-            if (versions.length > 0) {
-              const candidates = versions
-                .filter(v => semver.satisfies(v, versionSemVer))
-                .sort(semver.rcompare);
-              if (candidates.length > 0) {
-                resolve(candidates[0]);
-                return;
-              }
-            }
-          }
-          reject(
-            new CodedError(
-              'NoMatchingPackageVersion',
-              `No matching version of ${pkg}@${versionSemVer} found`,
-            ),
-          );
-        },
-      );
-    } else {
-      try {
-        exec(
-          `npm info ${pkg}@${versionSemVer} version --json`,
-          (err, stdout, _stderr) => {
-            try {
-              if (!err) {
-                let candidates = JSON.parse(stdout);
-                if (typeof candidates === 'string') {
-                  resolve(candidates);
-                  return;
-                }
-                candidates = candidates.sort(semver.rcompare);
-                if (candidates && candidates.length > 0) {
-                  resolve(candidates[0]);
-                  return;
-                }
-              }
-              reject(
-                new CodedError(
-                  'NoMatchingPackageVersion',
-                  `No matching version of ${pkg}@${versionSemVer} found`,
-                ),
-              );
-            } catch (e) {
-              reject(e);
-            }
-          },
-        );
-      } catch (err) {
-        reject(
-          new CodedError(
-            'NoMatchingPackageVersion',
-            `No matching version of ${pkg}@${versionSemVer} found`,
-          ),
-        );
+  const npmResponse = await npmFetch.json(pkg, {registry: NPM_REGISTRY_URL});
+
+  // Check if versionSemVer is a tag (i.e. 'canary', 'latest', 'preview', etc.)
+  if ('dist-tags' in npmResponse) {
+    const distTags = npmResponse['dist-tags'] as Record<string, string>;
+    if (versionSemVer in distTags) {
+      return distTags[versionSemVer];
+    }
+  }
+
+  // Check if versionSemVer is a semver version (i.e. '^0.60.0-0', '0.63.1', etc.)
+  if ('versions' in npmResponse) {
+    const versions = Object.keys(
+      npmResponse.versions as Record<string, unknown>,
+    );
+    if (versions.length > 0) {
+      const candidates = versions
+        .filter(v => semver.satisfies(v, versionSemVer))
+        .sort(semver.rcompare);
+      if (candidates.length > 0) {
+        return candidates[0];
```

Comment:
should we logverbose if there's more than one version?

Reply:
The point of the function is to find the latest version that fulfils the range specified, so I don't think that's necessary.